base/commandLine.cpp

```diff
@@ -1039,8 +1175,15 @@ bool processSettings(Common::String &command, Common::StringMap &settings, Commo
 		return true;
 	} else if (command == "auto-detect") {
 		// If auto-detects succeed, we want to return false so that the game is started
-		return !autoDetect(settings["path"]);
+		return !detectGames(settings["path"], true);
 		//return true;
+	} else if (command == "detect") {
+		detectGames(settings["path"], false);
+		return true;
```

Comment:
For example, for the --auto-detect command line. We want not only informations of detected games, but also to play the first one detected. 
So we `return !detectGames(settings["path"], true);` here, which means if we detect anything, return false->Run the game next. if we don't detect anything, return true->display information and do nothing else.